anesthetic/__init__.py

```diff
@@ -11,6 +11,34 @@
 import anesthetic.samples
 import anesthetic.plot
 
+import pandas
+import pandas.plotting._core
+import pandas.plotting._misc
+
+
+def _anesthetic_override(_get_plot_backend):
+    """Override the default backend.
+
+    When _get_plot_backend asks for 'matplotlib' it will be directed to
+    'anesthetic._matplotlib'. This is necessary since any users of
+    WeightedSamples should not be using the original backend.
+    """
+    def wrapper(backend=None):
+        if backend == 'matplotlib':
+            return _get_plot_backend('anesthetic._matplotlib')
+        return _get_plot_backend(backend)
+    return wrapper
+
+
+# Override the two places where _get_plot_backend is defined
+pandas.plotting._core._get_plot_backend = \
+        _anesthetic_override(pandas.plotting._core._get_plot_backend)
+pandas.plotting._misc._get_plot_backend = \
+        _anesthetic_override(pandas.plotting._misc._get_plot_backend)
+
+# Set anesthetic._matplotlib as the actual backend
+pandas.options.plotting.backend = 'anesthetic._matplotlib'
```

Comment:
This code sets up the anesthetic._matplotlib backend as a replacement for matplotlib. In principle it should behave exactly the same as the current matplotlib backend for normal pandas arrays, but behave correctly for `WeightedDataFrame`s